webapi/src/main/scala/org/knora/webapi/messages/admin/responder/usersmessages/UsersMessagesADM.scala

```diff
@@ -0,0 +1,767 @@
+                    permissions = PermissionsDataADM() // removed permissions
+                )
+            }
+            case UserInformationTypeADM.RESTRICTED => {
+
+                UserADM(
+                    id = id,
+                    email = email,
+                    password = None, // remove password
+                    token = None, // remove token
+                    givenName = givenName,
+                    familyName = familyName,
+                    status = status,
+                    lang = lang,
+                    groups = groups,
+                    projects = projects,
+                    sessionId = None, // removed sessionId
+                    permissions = permissions
+                )
+            }
+            case UserInformationTypeADM.FULL => {
+                UserADM(
+                    id = id,
+                    email = email,
+                    password = password,
+                    token = token,
+                    givenName = givenName,
+                    familyName = familyName,
+                    status = status,
+                    lang = lang,
+                    groups = groups,
+                    projects = projects,
+                    sessionId = sessionId,
+                    permissions = permissions
+                )
+            }
+            case _ => throw BadRequestException(s"The requested userTemplateType: $userTemplateType is invalid.")
+        }
+    }
+
+    /* Is the user a member of the SystemAdmin group */
+    def isSystemAdmin: Boolean = {
+        permissions.groupsPerProject.getOrElse(OntologyConstants.KnoraBase.SystemProject, List.empty[IRI]).contains(OntologyConstants.KnoraBase.SystemAdmin)
+    }
+
+    def isSystemUser: Boolean = id.equalsIgnoreCase(OntologyConstants.KnoraBase.SystemUser)
+
+    def isAnonymousUser: Boolean = id.equalsIgnoreCase(OntologyConstants.KnoraBase.AnonymousUser)
+
+    def fullname: String = givenName + " " + familyName
+
+    def getDigest: String = {
+        val md = java.security.MessageDigest.getInstance("SHA-1")
+        val time = System.currentTimeMillis().toString
+        val value = (time + this.toString).getBytes("UTF-8")
+        md.digest(value).map("%02x".format(_)).mkString
+    }
+
+    def setSessionId(sessionId: String): UserADM = {
+        UserADM(
+            id = id,
+            email = email,
+            password = password,
+            token = token,
+            givenName = givenName,
+            familyName = familyName,
+            status = status,
+            lang = lang,
+            groups = groups,
+            projects = projects,
+            permissions = permissions,
+            sessionId = Some(sessionId)
+        )
+    }
+
+    def isActive: Boolean = {
+        status
+    }
+
+    def toJsValue: JsValue = UsersADMJsonProtocol.userADMFormat.write(this)
+
+
+    // ToDo: Refactor by using implicit conversions (when I manage to understand them)
```

Comment:
I think you could use a value class: https://docs.scala-lang.org/overviews/core/value-classes.html

Reply:
thanks, I don't understand it but it looks good ;-)